openedx/core/djangoapps/content_libraries/views.py

```diff
@@ -61,20 +63,55 @@ def wrapped_fn(*args, **kwargs):
     return wrapped_fn
 
 
+class LibraryRootPagination(PageNumberPagination):
+    """
+    Paginates over ContentLibraryMetadata objects.
+    """
+    page_size = 50
+    page_size_query_param = 'page_size'
+
+
 @view_auth_classes()
 class LibraryRootView(APIView):
     """
     Views to list, search for, and create content libraries.
     """
 
+    @apidocs.schema(
+        parameters=[
+            apidocs.query_parameter(
+                'pagination',
+                bool,
+                description="Enables paginated schema",
+            ),
+            apidocs.query_parameter(
+                'page',
+                int,
+                description="Page number of result. Defaults to 1",
+            ),
+            apidocs.query_parameter(
+                'page_size',
+                int,
+                description="Page size of the result. Defaults to 50",
+            ),
+        ],
+    )
     def get(self, request):
         """
         Return a list of all content libraries that the user has permission to
         view. This is a temporary view for development and returns at most 50
         libraries.
         """
-        result = api.list_libraries_for_user(request.user)
-        return Response(ContentLibraryMetadataSerializer(result, many=True).data)
+        paginator = LibraryRootPagination()
+        queryset = api.get_libraries_for_user(request.user)
+        pageinated_qs = paginator.paginate_queryset(queryset, request)
```

Comment:
Nit: I think "paginate" is the correct verb (so, "paginated_qs", vs "pageinated_qs").